src/backend/executor/execExpr.c

```diff
@@ -1139,7 +1140,23 @@ ExecInitExprRec(Expr *node, ExprState *state,
 				 * value.
 				 */
 				scratch.opcode = EEOP_ROWIDEXPR;
-				scratch.d.rowidexpr.rowcounter = ((int64) GpIdentity.dbid) << 48;
+
+				/*
+				 * CBDB_PARALLEL
+				 * Planner have ensured that there is enough space for num of segments and parallel workers.
+				 * As we has not set ParallelWokerNumber yet now, use TotalParallelWorkerNumberOfSlice here
+				 * and keep bits space for ParallelWokerNumber.
+				 */
+				if (TotalParallelWorkerNumberOfSlice > 0)
+				{
+					int parallel_bits = pg_leftmost_one_pos32(TotalParallelWorkerNumberOfSlice) + 1;
+					int segs = getgpsegmentCount();
```

Comment:
Both use getgpsegmentCount or dbid?

Reply:
Good idea!
Instead of total segments, use dbid for each query, we could make more room for parallel worker numbers.
If a query on a database with a small dbid, there could be more parallel workers for it~

Reply:
Oh, wait, there is a thinko.
We should use getgpsegmentCount() to check room, because on each segments , dbid is different.
But we should make sure all segments of a plan have enough room for segbits, so getgpsegmentCount is used for that.

Reply:
In fact, segs here  is a double check as we have already checked it during planner.
This check could be removed.

Reply:
AFAK, use a Max(DBID) is the right one in case that some segments down and recovered or CBDB drop some segments and then expanded.
@my-ship-it

Reply:
> AFAK, use a Max(DBID) is the right one in case that some segments down and recovered or CBDB drop some segments and then expanded.

 Codes updated as that.